polls/views.py

```diff
@@ -1,11 +1,34 @@
-from django.http import HttpResponse
+from django.views import View
 from django.shortcuts import render
-from django.template import loader
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.views.generic import TemplateView
+from django.utils import timezone
 
-# Create your views here.
+class IndexView(TemplateView):
+    template_name = "polls/index.html"
 
+class AuthFlowView(LoginRequiredMixin, View):
+    template_name = "polls/auth_flow.html"
+    min_time = 900
+    max_time = 4500
+    def get(self, request):
+        if not request.user.first_task_of_this_session_performed_at:
+            # if it's user's first time
+            context = {
+                "should_rest": False,
+                "can_continue": False
+                }
+            return render(request, self.template_name, context)
 
-def index(request):
-    template = loader.get_template("polls/index.html")
-    context = {}
-    return HttpResponse(template.render(context, request))
+        current_time = timezone.now()
+        time_diff = current_time - request.user.first_task_of_this_session_performed_at
+        time_diff_seconds = time_diff.total_seconds()
```

Comment:
then total_minutes

Reply:
`total_minutes()` is not available for `time_delta` but I have added manual conversion for minutes.